File: Dart/old_approach/old_approach.dart
import 'dart:io';

class Reverse {
	int reverseNumber(int number) {
		var digit = 0;
		var sum = 0;

		while (number != 0) {
			digit = number % 10;
			sum = sum * 10 + digit;
			number = number ~/ 10;
		}

		return sum;
	}
}

void main() {
	Reverse rev = new Reverse();
	var input = stdin.readLineSync();
	var number = int.parse(input);
	var res = rev.reverseNumber(number);
	print(res);
}

File: Go/old_approach/old_approach.go
package main

import (
	"fmt"
	"strconv"
)

type Reverse struct {
	number int
}

func(rev Reverse) ReverseNumber() int {
	var digit int = 0
	var sum int = 0
	var num int = rev.number

	for num != 0 {
		digit = num % 10
		sum = sum * 10 + digit
		num /= 10
	}

	return sum
}

func main() {
	var input string
	fmt.Scanln(&input)
	
	num, _ := strconv.Atoi(input)

	rev := Reverse{number: num}
	var res int = rev.ReverseNumber()
	fmt.Println(res)
}